src/mock_crust/support.rs

```diff
@@ -360,6 +369,12 @@ impl ServiceImpl {
     }
 
     fn handle_disconnect(&mut self, peer_endpoint: Endpoint) {
+        // Drop any remaining messages between these two nodes. We do this now,
+        // not when `disconnect()` is called, to allow the disconnect message
+        // and any prior messages to reach their destinations.
```

Comment:
But real Crust probably doesn't even do that - I don't think it sends a `Disconnect` message; it probably just closes the connection -, so it might make sense to actually drop these in `disconnect()`.